### PR TITLE
Whitelist mandatory coinbase destination

### PR DESCRIPTION
**Pull request by lawlawlaw**
The mandatory coinbase destination is added to the address whitelist.
This is so that all outputs can be paid to the 'fees' address.

### CODE REVIEW

File: src/init.cpp [hunk 1]
```diff
@@ -1222,6 +1222,7 @@ bool AppInitSanityChecks()
 bool AppInitMain(boost::thread_group& threadGroup, CScheduler& scheduler)
 {
     const CChainParams& chainparams = Params();
+    addressWhitelist.InitCoinbaseDest();
```
Comment (nkostoulas) [inline src/init.cpp]: This should only be done if we are using whitelisting.

File: src/policy/whitelist.cpp [hunk 2]
```diff
@@ -846,10 +858,14 @@ void CWhiteList::clear(){
...
   boost::recursive_mutex::scoped_lock scoped_lock(_mtx);
-  if(!find(keyId)) return false;
-  if(_kycMap[keyId] == CKeyID()) return true;
-  if(!find_kyc_whitelisted(_kycMap[keyId])) return false;
-  return true;
+  if(find(keyId)) {
+    if(_kycMap[keyId] == CKeyID()) return true;
+    if(!find_kyc_whitelisted(_kycMap[keyId])) return false;
+  }
+  if(_coinbase_dest != nullptr){
```
Comment (nkostoulas) [inline src/policy/whitelist.cpp]: Why not just add coinbase on init and then nothing changes in is_whitelisted(), not need for pointers, nothing.

File: src/init.cpp [hunk 3]
```diff
@@ -1222,6 +1222,20 @@ bool AppInitSanityChecks()
 bool AppInitMain(boost::thread_group& threadGroup, CScheduler& scheduler)
 {
     const CChainParams& chainparams = Params();
+    
+    if (fRequireWhitelistCheck  || fScanWhitelist){
```
Comment (nkostoulas) [inline src/init.cpp]: I'd move this further down where there is already a fRequireWhitelistCheck, up to you.

File: src/test/test_bitcoin.cpp [hunk 4]
```diff
@@ -90,6 +94,7 @@ TestingSetup::TestingSetup(const std::string& chainName, const std::string& fedp
         pathTemp = GetTempPath() / strprintf("test_bitcoin_%lu_%i", (unsigned long)GetTime(), (int)(GetRand(100000)));
         boost::filesystem::create_directories(pathTemp);
         ForceSetArg("-datadir", pathTemp.string());
+        ForceSetArg("-con_mandatorycoinbase", std::string("76a914567884b53d417d36b37a0409521f4644a7f46ffe88ac"));
```
Comment (nkostoulas) [inline src/test/test_bitcoin.cpp]: This might be obsolete.

**Comment by lawlawlaw** [comment]
Have done this in a better way - PR on the way.

**Pull request closed without merging**